src/main/java/uk/gov/hmcts/reform/managecase/service/NoticeOfChangeService.java

```diff
@@ -262,6 +274,46 @@ public RequestNoticeOfChangeResponse requestNoticeOfChange(NoCRequestDetails noC
             .build();
     }
 
+    public void checkNoticeOfChangeApproval(String caseId) {
+        CaseViewResource caseViewResource = getCase(caseId);
+        CaseViewEvent[] events = caseViewResource.getCaseViewEvents();
+
+
+        if (events.length != EXPECTED_NUMBER_OF_EVENTS) {
+            throw new ValidationException(NOC_DECISION_EVENT_UNIDENTIFIABLE);
+        }
+
+        String eventId = events[0].getEventId();
+
+        CaseUpdateViewEvent caseUpdateViewEvent = dataStoreRepository.getStartEventTrigger(caseId, eventId);
+
+        if (caseUpdateViewEvent != null) {
+            if (caseUpdateViewEvent.getEventToken() == null) {
+                throw new ValidationException(EVENT_TOKEN_NOT_PRESENT);
+            }
+
+            Map<String, JsonNode> data =
+                caseUpdateViewEvent.getCaseFields().stream()
+                    .filter(cvf -> cvf.getFieldTypeDefinition().getId().equals(PREDEFINED_COMPLEX_ORGANISATION_POLICY)
+                        || cvf.getFieldTypeDefinition().getId().equals(PREDEFINED_COMPLEX_CHANGE_ORGANISATION_REQUEST))
+                    .collect(Collectors.toMap(
+                        CaseViewField::getId, cvf -> jacksonUtils.convertValue(cvf, JsonNode.class)));
+
+
+            Event event = Event.builder()
+                .eventId(eventId)
+                .description(CHECK_NOC_APPROVAL_DESCRIPTION)
+                .build();
+
+            CaseDataContent caseDataContent = CaseDataContent.builder()
+                .token(caseUpdateViewEvent.getEventToken())
+                .event(event)
+                .data(data)
+                .build();
+            dataStoreRepository.submitEventForCaseOnly(caseId, caseDataContent);
```

Comment:
you need to inject NocApprovalDataStoreRepository in this case.

Reply:
Please elaborate...